prototype/FFT_image_processing/bitmap.c

```diff
@@ -0,0 +1,268 @@
+#include "bitmap.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct ImageProperties {
+  unsigned int height;
+  unsigned int width;
+  unsigned int paddingSize;
+  unsigned int bytesPerPixel;
+} ImageProperties;
+
+static void writeBitmapHeaderRGB(const ImageProperties properties,
+                                 FILE* imageFile);
+static void writeBitmapHeaderGrey(const ImageProperties properties,
+                                  FILE* imageFile);
+static void writeGenericHeader(const unsigned int numberOfColors,
+                               const ImageProperties properties,
+                               FILE* imageFile);
+static inline unsigned int getPaddingSize(const unsigned int width,
+                                          const unsigned int bytesPerPixel);
+static void writeImageToFile(const unsigned char* image,
+                             const ImageProperties properties, FILE* imageFile);
+
+/**
+ * @brief Transforms the incoming RGB channels to Greyscale
+ * @param imageData The image data that will be transformed to Greyscale.
+ * Expected channel format: B, G, R (default bitmap format)
+ * @param numberOfPixels The number of pixels of the incoming RGB image data
+ * @return A 8-bit channel containing the Greyscale value
+*/
+unsigned char* convertRGBtoGrey(unsigned char* imageData, const unsigned int numberOfPixels) {
+  
+  unsigned char* greyData = malloc(numberOfPixels * GREY_BYTES * sizeof(unsigned char));
```

Comment:
Hmm, I see a malloc here but no free. Does it mean the end user is responsible for freeing the memory ?

Reply:
Yes. A comment was added.

Reply:
Thank you.